addons/web/static/src/legacy/js/core/dialog.js

```diff
@@ -238,6 +238,13 @@ var Dialog = Widget.extend({
      *   `on_close` handler.
      */
     destroy: function (options) {
+        // If there are more than one modal shown do not remove the
+        // 'prevent-spinner' class
+        const numberOfModalShown = document.querySelectorAll(".modal_shown").length;
+        const spinnerEl = document.querySelector(".o_we_ui_loading");
+        if (spinnerEl && numberOfModalShown === 1) {
+            spinnerEl.classList.remove("prevent-spinner");
+        }
```

Comment:
If you do like what I said in my previous comment, these lines will surely be removed :wink: 
But in any case, maybe check for `numberOfModalShown <= 1` instead, to be more robust ? :slightly_smiling_face: